src/include/network/abstract_command_factory.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+#include <memory>
+#include "network/abstract_network_command.h"
+
+#define MAKE_COMMAND(type) std::static_pointer_cast<AbstractNetworkCommand, type>(std::make_shared<type>(packet))
+
+namespace terrier::network {
+
+/**
+ * A command factory takes in input packets and converts them to commands usable by the protocol interpreters
+ */
+class AbstractCommandFactory {
+ public:
+  /**
+   * Convert a packet to a command.
+   * @param packet the input packet
+   * @return a shared_ptr to the converted command
+   */
+  virtual std::shared_ptr<AbstractNetworkCommand> PacketToCommand(InputPacket *packet) = 0;
+
+  virtual ~AbstractCommandFactory() = default;
+};
```

Comment:
I am not sure why this is needed as the abstraction layer drawn to the rest of the network layer is `ProtocolInterpreter`. Do you use the type `AbstractCommandFactory` anywhere in the codebase where it actually exhibits polymorphism?

Reply:
Both `PostgresCommandFactory` and `ITPCommandFactory` inherit from this class. I am not sure what future APIs we will need in this abstract though.

Reply:
In that this case I would not implement this abstract class now as it serves no purpose.

Reply:
@GustavoAngulo